tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -119,13 +116,14 @@ private static OpenSpan startSpanInternal(String operation, SpanType type) {
                 .spanId(Tracers.randomId())
                 .type(type);
 
-        Optional<OpenSpan> prevState = currentTrace.get().top();
+        Trace trace = getOrCreateCurrentTrace();
+        Optional<OpenSpan> prevState = trace.top();
         if (prevState.isPresent()) {
             spanBuilder.parentSpanId(prevState.get().getSpanId());
         }
```

Comment:
nit: could simplify

```suggestion
        trace.top().ifPresent(openSpan -> spanBuilder.parentSpanId(openSpan.getSpanId()));
```

Reply:
I opted to avoid the lambda allocation since this is a relatively hot path. We do a lot elsewhere, so I could go either way.

Reply:
makes sense